corehq/apps/userreports/views.py

```diff
@@ -535,12 +536,15 @@ def dispatch(self, request, *args, **kwargs):
         self._populate_data_source_properties_from_interface(data_source_interface)
         return super(ConfigureReport, self).dispatch(request, *args, **kwargs)
 
-    def render_error_response(self, message):
+    def render_error_response(self, message, allow_delete=None):
         if self.existing_report:
-            context = {'report_id': self.existing_report.get_id,
-                       'is_static': self.existing_report.is_static}
+            context = {
+                'allow_delete': self.existing_report.get_id and not self.existing_report.is_static
+            }
         else:
             context = {}
+        if allow_delete is not None:
```

Comment:
Nitpick: this code would behave the same without the if statement, wouldn't it?

Reply:
no, without the `if` it would always put the argument value into the context which would override any value already in the context.